casper/src/main/scala/coop/rchain/casper/blocks/BlockReceiver.scala

```diff
@@ -0,0 +1,280 @@
+package coop.rchain.casper.blocks
+
+import cats.effect.concurrent.Ref
+import cats.effect.{Concurrent, Sync}
+import cats.syntax.all._
+import coop.rchain.blockstorage.BlockStore
+import coop.rchain.blockstorage.BlockStore.BlockStore
+import coop.rchain.blockstorage.dag.BlockDagStorage
+import coop.rchain.casper.engine.BlockRetriever
+import coop.rchain.casper.protocol.BlockMessage
+import coop.rchain.casper.syntax._
+import coop.rchain.casper.util.ProtoUtil
+import coop.rchain.casper.{CasperShardConf, PrettyPrinter, Validate}
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.shared.Log
+import coop.rchain.shared.syntax._
+import fs2.Stream
+import fs2.concurrent.Queue
+
+sealed trait RecvStatus
+// Checking if block should be ignored or continue with processing
+case object BeginReceived extends RecvStatus
+// Block received and stored in blocks store, waiting for validation and DAG insertion
+case object ReceivedAndStored extends RecvStatus
+// Block sent to validation
+case object PendingValidation extends RecvStatus
+// Requested missing dependencies
+case object PendingRequest extends RecvStatus
```

Comment:
I'm thinking as BlockReceiver component purpose is to `receive` blocks then the final state of the blocks should be `Received`. But what does this `received` mean - should be explained inside a component. And in our case `received` means that block should be persisted and all dependencies of a block should be validated. 

Would that make sense then to make the chain of statuses
Accepted -> Stored -> Received
and make `beginReceived ` -> `accept`
`endReceived ` -> `store`
This `endReceived` is to me a bit confusing, because it tells that job is done here, but it is actually not.